add GenderCardView

AdrianLunaVillegas opened the pull request:
create a gendercardview to optimize the code

AdrianLunaVillegas committed:
add GenderCardView

create a gendercardview to optimize the code

xavijimenezmulet commented on app/src/main/AndroidManifest.xml:
```diff
@@ -34,10 +34,12 @@
             android:name=".firstapp.common.GenderCardView"
-            android:exported="false"/>
+            android:exported="false" />
+
```
You should use just a one blank space.

xavijimenezmulet commented on app/src/main/java/com/adriandevs/androidmaster1/firstapp/common/GenderCardView.kt:
```diff
@@ -3,33 +3,27 @@ package com.adriandevs.androidmaster1.firstapp.common
-
-        imageView.setImageResource(customImage)
-        textView.text = customText
+        binding.apply {
```
Put this in a function, some like initData()

xavijimenezmulet commented on app/src/main/res/layout/activity_gender.xml:
Why is this unused xml?

jose-ojalvo commented on app/src/main/res/layout/card_view_gender.xml:
```diff
@@ -1,52 +1,21 @@
     android:orientation="vertical">
 
+    <ImageView
+        android:id="@+id/gMale"
```
View IDs should describe briefly the purpose of the view, avoid describing the use case nor the implementation. 

Please modify the IDs to match the usage (e.g genderImage instead of gMale)

jose-ojalvo commented on app/src/main/res/layout/card_view_gender.xml:
```diff
@@ -1,52 +1,21 @@
-
-    </androidx.core.widget.NestedScrollView>
+    <TextView
+        android:id="@+id/tvMale"
```
Same as stated above, avoid using the use case of the component for the naming of its views, it will lead to misleading use of this view.

jose-ojalvo commented on app/src/main/res/values/attrs.xml:
```diff
@@ -2,10 +2,8 @@
-    <declare-styleable name="GenderCardView_customText"></declare-styleable>
-    <attr name="customImage" format="reference" />
-    <attr name="customText" format="string" />
+        <attr name="customImage" format="reference" />
```
Just as a warning, don't use generic parameter names in the styleable attributes because the names cannot be reused in another style declaration.

Take that into account when naming attributes (you won't be able to use "customImage" in another style).

jose-ojalvo commented on app/src/main/AndroidManifest.xml:
```diff
@@ -34,10 +34,12 @@
         <activity
             android:name=".firstapp.common.GenderCardView"
-            android:exported="false"/>
+            android:exported="false" />
```
Why did you add a custom view as an activity in the manifest??
Also, does your activities allow orientation switching? Be mindful of that behavior as it can break your app

AdrianLunaVillegas committed:
add

AdrianLunaVillegas committed:
empty commit / fixing the broken

AdrianLunaVillegas committed:
Merge branch 'feaure/genderCardView' of https://github.com/AdrianLunaVillegas/first-project-android into feaure/genderCardView

jose-ojalvo reviewed (changes requested):
There are still changes to be made in order to merge this pull request to master, please take a look at what's still remaining

AdrianLunaVillegas committed:
reviewind and fixing the pull request

There is still an error missing, which Xavi gave me.

AdrianLunaVillegas committed:
light commit

I messed up something by changing the name and I fixed it, and I'm starting a new app "TODOApp"

AdrianLunaVillegas committed:
finishing previous application and  starting the new APP

In the next commit I will separate by branches.
In this new app I am learning to use recyclersview.

xavijimenezmulet commented on app/src/main/java/com/adriandevs/androidmaster1/todoapp/CategoriesViewHolder.kt:
```diff
@@ -0,0 +1,41 @@
+
+    fun render(taskCategory: TaskCategory) {
+        viewBinding.apply {
+            tvCategoryName.text = "EXEMPLE"
```
Why is this string is hardcoded?

xavijimenezmulet commented on app/src/main/java/com/adriandevs/androidmaster1/todoapp/TasksAdapter.kt:
```diff
@@ -0,0 +1,21 @@
+
+class TasksAdapter(private val tasks: List<Task>) : RecyclerView.Adapter<TasksViewHolder>() {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TasksViewHolder {
+        val view =
```
Why you are not using view binding ?

xavijimenezmulet commented on app/src/main/java/com/adriandevs/androidmaster1/todoapp/TasksViewHolder.kt:
```diff
@@ -0,0 +1,10 @@
+
+class TasksViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    fun render(task: Task) {
+
```
Why it's empty ?

AdrianLunaVillegas committed:
creating the third app

I have used the same branch again, sorry!
it will not happen again
